server/src/main/java/org/opensearch/rest/action/admin/cluster/RestClusterPutWeightedRoutingAction.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.cluster.shards.routing.weighted.put.ClusterPutWeightedRoutingRequest;
+import org.opensearch.client.Requests;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/**
+ * Update Weighted Round Robin based shard routing weights
+ *
+ * @opensearch.api
+ *
+ */
+public class RestClusterPutWeightedRoutingAction extends BaseRestHandler {
+
+    private static final Logger logger = LogManager.getLogger(RestClusterPutWeightedRoutingAction.class);
+
+    @Override
+    public List<Route> routes() {
+        return singletonList(new Route(PUT, "/_cluster/routing/awareness/{attribute}/weights"));
```

Comment:
Do we really need `attribute` in the route? Can it be included in the payload instead?

Reply:
Yeah, we had multiple discussion on this. Finalize on this one. More details can be found here - [feature/wrr-put-api](https://github.com/opensearch-project/OpenSearch/issues/3639#issuecomment-1219776875)